gpkit/constraints/relax.py

```diff
@@ -0,0 +1,140 @@
+                        self.relaxvar >= 1],
+                       substitutions,
+                       name=False)
+
+
+class RelaxConstraints(Model):
+    """Relax constraints optimally, as in Eqn. 11 of [Boyd2007].
+
+    Arguments
+    ---------
+    constraints : iterable
+        Constraints which will be relaxed (made easier).
+
+
+    Attributes
+    ----------
+    relaxvars : Variable
+        The variables controlling the relaxation. A solved value of 1 means no
+        relaxation was necessary or optimal for a particular constraint.
+        Higher values indicate the amount by which that constraint has been
+        made easier: e.g., a value of 1.5 means it was made 50 percent easier
+        in the final solution than in the original problem.
+
+    [Boyd2007] : "A tutorial on geometric programming", Optim Eng 8:67-122
+
+    """
+
+    def __init__(self, constraints):
+        if not isinstance(constraints, ConstraintSet):
+            constraints = ConstraintSet(constraints)
+        substitutions = dict(constraints.substitutions)
+        posynomials = constraints.as_posyslt1()
+        N = len(posynomials)
+        self.relaxvars = VectorVariable(N, "C", models=["Relax"],
+                                        modelnums=[Model._nums["Relax"]])
+        Model.__init__(self, self.relaxvars.prod(),
+                       [[posynomials <= self.relaxvars],
+                        self.relaxvars >= 1],
+                       substitutions,
+                       name=False)
+
+
+class RelaxConstants(Model):
+    """Relax constraints optimally, as in Eqn. 11 of [Boyd2007].
+
+    Arguments
+    ---------
+    constraints : iterable
+        Constraints which will be relaxed (made easier).
+
+    include_only : set
+        if declared, variable names must be on this list to be relaxed
+
+    exclude : set
+        if declared, variable names on this list will never be relaxed
+
+
+    Attributes
+    ----------
+    relaxvars : Variable
+        The variables controlling the relaxation. A solved value of 1 means no
+        relaxation was necessary or optimal for a particular constant.
+        Higher values indicate the amount by which that constant has been
+        made easier: e.g., a value of 1.5 means it was made 50 percent easier
+        in the final solution than in the original problem. Of course, this
+        can also be determined by looking at the constant's new value directly.
+    """
+    def __init__(self, constraints, include_only=None, exclude=None):
+        exclude = frozenset(exclude) if exclude else frozenset()
+        include_only = frozenset(include_only) if include_only else frozenset()
+        if not isinstance(constraints, ConstraintSet):
+            constraints = ConstraintSet(constraints)
+        substitutions = dict(constraints.substitutions)
+        constants, _, _ = parse_subs(constraints.varkeys,
+                                     constraints.substitutions)
```

Comment:
the include_only/exclude code? Sure, it's a design pattern that I want to standardize.